pkg/proxy/iptables/proxier.go

```diff
@@ -169,6 +169,8 @@ type Proxier struct {
 	haveReceivedServiceUpdate   bool // true once we've seen an OnServiceUpdate event
 	haveReceivedEndpointsUpdate bool // true once we've seen an OnEndpointsUpdate event
 	throttle                    flowcontrol.RateLimiter
+	timer                       *time.Timer
+	lastSync                    time.Time
```

Comment:
I think avioiding endpoint spam should be pretty easy without an extensive refactor. The recommended code structure is 100% limited to this pr, not the surrounding code, so we don't make things any harder to debug.